### PR TITLE
Add azure: build_id to schema

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by jaimergp** [comment]
Nice! Should we also restore `azure.user_or_org` pre-schema behavior?

File: conda_smithy/schema.py [hunk 1]
```diff
@@ -172,6 +172,11 @@ class AzureConfig(BaseModel):
         description="The name of the Azure Pipelines project",
     )
 
+    build_id: Optional[int] = Field(
+        default=None,
+        description="The Build ID for the specific feedstock",
```
Comment (beckermr) [inline conda_smithy/schema.py:177]: Since this will appear in our docs, can we add an explanation of what this does and why one might use it?

**Comment by isuruf** [comment]
> Nice! Should we also restore azure.user_or_org pre-schema behavior?

It's already there.

**Comment by isuruf** [comment]
pre-commit.ci autofix

**Review by beckermr — APPROVED** [review]

**Comment by jdblischak** [comment]
> It's already there.

Nice. I confirmed locally that `azure.user_or_org` this works to update the URLs in the README.

But the description is confusing. It says GitHub when it should say Azure

https://github.com/conda-forge/conda-smithy/blob/44acb937c50ce98adf1b50b01a6ab7bc577d2fb2/conda_smithy/data/conda-forge.json#L136

**Review by jdblischak — APPROVED** [review]
I built this branch locally and confirmed that the edits I made to `azure.build_id`, `azure.user_or_org`, and `azure.project_name` were reflected in the README Azure URLs. Thanks @isuruf!

**Review by jaimergp — APPROVED** [review]

**Pull request merged**